src/common/tests/CommonUtilsUT.cpp

```diff
@@ -642,35 +642,6 @@ TEST_F(CommonUtilsTest, ExecuteLongCommand)
     FREE_MEMORY(textResult);
 }
 
-TEST_F(CommonUtilsTest, ExecuteTooLongCommand)
-{
-    char* textResult = nullptr;
-
-    size_t commandLength = (size_t)(sysconf(_SC_ARG_MAX) + 1);
-    char* command = (char*)malloc(commandLength);
-    EXPECT_NE(nullptr, command);
-
-    if (nullptr != command)
-    {
-        snprintf(command, commandLength, "echo ");
-        size_t echoLength = strlen(command);
-        EXPECT_EQ(5, echoLength);
-
-        for (size_t i = echoLength; i < (commandLength - 1); i++)
-        {
-            command[i] = (i % 2) ? '0' : '1';
-        }
-        command[commandLength - 1] = 0;
-
-        EXPECT_EQ(E2BIG, ExecuteCommand(nullptr, command, true, true, 0, 0, &textResult, nullptr, nullptr));
-        EXPECT_EQ(nullptr, textResult);
-
-        free(command);
-    }
-
-    FREE_MEMORY(textResult);
-}
```

Comment:
Let's keep this unit-test. We know to expect E2BIG when the command is too long. We should keep UT-ing this.

Why:

E2BIG is not a normal command failure. It's our failure because we passed in a too long command. It's something that we need to watch for and if this ever happens at run-time, we need to treat this case as a true error case for us and a code bug, and not just a normal command result.

Personally I would also put back the validation against _SC_ARG_MAX early in the ExecuteCommand call, validate, log error and fail early, before forking and rest.

Reply:
Realistically speaking - what are the chances of getting ExecuteCommand with a 2 megabyte long command?

Reply:
The function cand also execute whole scripts, not just single commands. There could also be a bug with a non-terminated command string.

Reply:
Done.